### PR TITLE
Tag children add multiple entities

### PR DESCRIPTION
**Pull request by spitfire305**

### CODE REVIEW

File: app/Http/Controllers/Tags/ChildController.php [hunk 1]
```diff
@@ -79,8 +79,8 @@ public function store(StoreTagEntity $request, Campaign $campaign, Tag $tag)
             $redirectUrlOptions['tag_id'] = $tag->id;
         }
 
-        $tag->attachEntity($request->only('entity_id'));
+        $count = $tag->attachEntity($request->only('entities'));
```
Comment (ilestis) [inline app/Http/Controllers/Tags/ChildController.php:82]: attachEntitie__s__() please :)

File: app/Models/Tag.php [hunk 2]
```diff
@@ -292,26 +292,28 @@ public function hasColour(): bool
...
             }
-        }
 
-        $entityTag = EntityTag::create([
-            'tag_id' => $this->id,
-            'entity_id' => $entityId
-        ]);
-
-        return $entityTag !== false;
+            EntityTag::create([
```
Comment (ilestis) [inline app/Models/Tag.php:310]: Can we just do `$entity->tags->syncWithoutDetatching([$this->id])` ?

Comment (spitfire305) [reply app/Models/Tag.php:310]: syncWithoutDetatching() doesnt work with hasManyThrough relations.

File: resources/views/components/forms/foreign.blade.php [hunk 3]
```diff
@@ -19,9 +22,14 @@ class="w-full select2 join-item"
...
             data-allow-clear="{{ $allowClear ? 'true' : 'false' }}"
             @if (!empty($dropdownParent)) data-dropdown-parent="{{ $dropdownParent }}" @endif
+            :dropdownParent="request()->ajax() ? '#primary-dialog' : null"
     >
         @foreach ($options as $key => $value)
-            <option value="{{ $key }}">{!! $value !!}</option>
+            @if ($multiple)
+                <option value="{{ $key }}">{!! $value !!}</option>
+            @else
+                <option value="{{ $key }}" class="select2-entity" selected="selected">{{ $entity->name }}</option>
```
Comment (ilestis) [inline resources/views/components/forms/foreign.blade.php:31]: Why $entity->name ? Where is that value coming from magically? Why the select2-entity class? When editing a location, does this change break the parent location dropdown?

File: resources/views/tags/entities/_form.blade.php [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
-            :campaign="$campaign"
-            :ajax="request()->ajax()"
-        >
-        </x-form.entities>
-    </x-forms.field>
+    <x-forms.foreign
+        field="entities"
+        :required="true"
+        label="abilities.show.tabs.entities"
+        multiple="multiple"
```
Comment (ilestis) [inline resources/views/tags/entities/_form.blade.php:8]: I feel like multiple should be a boolean, and validated on the Foreign.php component

File: resources/views/components/forms/foreign.blade.php [hunk 5]
```diff
@@ -10,7 +10,10 @@
         <div class="join w-full">
     @endif
 
-    <select name="{{ $name }}" id="{{ $id }}"
+    <select 
+            multiple="$multiple"
```
Comment (ilestis) [inline resources/views/components/forms/foreign.blade.php:14]: image
I can now set multiple locations on a family

**Review by ilestis — CHANGES_REQUESTED** [review]
Broke all other dropdowns

**Review by ilestis — APPROVED** [review]
:chefkiss:

**Pull request merged**